pylint/reporters/reports_handler_mix_in.py

```diff
@@ -24,12 +24,12 @@ def __init__(self) -> None:
         self._reports: ReportsDict = collections.defaultdict(list)
         self._reports_state: Dict[str, bool] = {}
 
-    def report_order(self) -> List[checkers.BaseChecker]:
+    def report_order(self) -> List["BaseChecker"]:
```

Comment:
The changes in this file are the only once not directly related to `PyLinter`. It would be better to deal with them separately. Please create another PR for it, especially considering the size fo this one such a change would get lost otherwise.

Reply:
Originally it was necessary to make mypy works, we did multiple prior MR already (#5564 #5574).

Reply:
Overlooked that it was just an import change. Btw. here the `if TYPE_CHECKING` guard does make sense as it's a mixin module for `pylinter` itself.